### PR TITLE
Custom predictors

### PR DESCRIPTION
**Pull request by kkannan7**
Created a feature to accept custom predictors in prog_server. 
- The test code is written into integration.py
- The test code tries to use the Monte Carlo from progpy and succeeds

### CODE REVIEW

File: tests/integration.py [hunk 1]
```diff
@@ -243,6 +245,40 @@ def test_custom_models(self):
     def tearDownClass(cls):
         prog_server.stop()
 
+    def test_custom_predictors(self):
+        # Restart server with model
+        prog_server.stop()
+        #define the model
+        ball = ThrownObject(thrower_height=1.5, throwing_speed=20)
+        prog_server.start(models={'ball': ball}, port=9883)
```
Comment (teubert) [inline tests/integration.py:253]: This doesn't test the new feature. We need to pass the predictor into start function like we do with model here.

Then when we create a session we have to tell it to use the new predictor (which is actually just a Monte Carlo)

**Review by teubert — CHANGES_REQUESTED** [review]

**Review by kjjarvis — CHANGES_REQUESTED** [review]
This looks great. Nice addition. Just a few small adjustments and then I think it's ready

**Review by kjjarvis — COMMENTED** [review]
Great edits. One more suggestion - as per our conversation, let's add a test to check the TypeError that's in line 38 of prog_server

**Review by kjjarvis — APPROVED** [review]
Nice addition of custom predictors and corresponding tests

**Pull request merged**